src/main/java/fr/insee/queen/api/configuration/SwaggerConfiguration.java

```diff
@@ -0,0 +1,106 @@
+* 3 ways to authenticated : 
+* 	- without authentication,
+* 	- basic authentication 
+* 	- and keycloak authentication 
+* 
+* @author Claudel Benjamin
+* 
+*/
+@Configuration
+@EnableSwagger2
+public class SwaggerConfiguration {
+	/**
+	 * The name of Spring application.<br>
+	 * Generate with the application property spring.application.name
+	 */
+    @Value("${spring.application.name}")
+    private String name;
+    
+    @Bean
+    public Docket api() {
+    	List<ResponseMessage> messages = new ArrayList<ResponseMessage>();
+    	messages.add(new ResponseMessageBuilder().code(200).message("Success!").build());
+    	messages.add(new ResponseMessageBuilder().code(401).message("Not authorized!").build());
+    	messages.add(new ResponseMessageBuilder().code(403).message("Forbidden!").build());
+    	messages.add(new ResponseMessageBuilder().code(404).message("Not found!").build());
+    	String urlString = "/" + name;
+        Docket docket = new Docket(DocumentationType.SWAGGER_2)
+        		.useDefaultResponseMessages(false)
+        		.globalResponseMessage(RequestMethod.GET, messages)
+        		.globalResponseMessage(RequestMethod.POST, messages)
+        		.globalResponseMessage(RequestMethod.PUT, messages)
+                .apiInfo(apiInfo())
+                .ignoredParameterTypes(HttpServletRequest.class, HttpServletResponse.class, HttpSession.class)
+                .alternateTypeRules(AlternateTypeRules.newRule(StreamingResponseBody.class, MultipartFile.class));
+        try {
+            URL url = new URL(urlString);
+            docket = docket.pathMapping(url.getPath()).host(url.getHost() + (url.getPort() > -1 ? ":" + url.getPort() : ""));
+        } catch (MalformedURLException e) {
+            docket = docket.pathMapping(urlString);
+        }
+        return docket
+                .pathProvider(absolutePathProvider())
+                .securitySchemes(List.of(new BasicAuth(name)))
+                .securityContexts(List.of(SecurityContext.builder()
+                		.securityReferences(List.of(new SecurityReference(name, new AuthorizationScope[0])))
+                		.forPaths(PathSelectors.regex("/.*")).build()))
+                .select()
+                .apis(RequestHandlerSelectors.any())
+                .paths(PathSelectors.regex("/.*"))
+                .build();
+    }
+    
+    private PathProvider absolutePathProvider() {
+        return new AbstractPathProvider() {
+            @Override
+            protected String getDocumentationPath() {
+                return "";
+            }
+            @Override
+            protected String applicationPath() {
+                return "";
+            }
+        };
+    }
+    
+    private ApiInfo apiInfo() {
+        return new ApiInfo("Queen-Back-Office", "Back-office services for Queen", "1.0", "", new Contact("Metallica", "", "dr59-sndi-metallica@outlook.fr"), "LICENSEE", "https://github.com/InseeFr/Queen-Back-Office/blob/master/LICENSE", List.of());
```

Comment:
Remove mail contact for Metallica, prefer link to gihtub issues.

Reply:
fixed